processing/src/main/java/org/apache/carbondata/processing/sort/sortdata/SingleThreadFinalSortFilesMerger.java

```diff
@@ -135,8 +135,8 @@ public void addInMemoryRawResultIterator(List<RawResultIterator> sortedRawResult
               noDicAndComplexColumns, sortParameters, measureDataType);
       if (inMemorySortTempChunkHolder.hasNext()) {
         inMemorySortTempChunkHolder.readRow();
+        recordHolderHeapLocal.add(inMemorySortTempChunkHolder);
```

Comment:
`inMemorySortTempChunkHolder`  ->  adapter for RawResultIterator on sorted files
`recordHolderHeapLocal`  ->  Priority Queue

This line adds `inMemorySortTempChunkHolder` to the `recordHolderHeapLocal` so that it can be used for final merge sorting.

Reply:
still cannot understand this change?
Is it for a bug?
If yes, please mention that in the PR description about the scenario and the root cause and add test case for that. -- I'd recommend you to raise a new PR for it.

Reply:
Yes. It is for a bug. Add the `inMemorySortTempChunkHolder` object in the priority queue only if data is present in that object. Earlier it was adding it anyway. This fixes that mistake.